bridge/client/app/_components/ktb-time-input/ktb-time-input.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Timeframe } from '../../_models/timeframe';
+import { FormControl, FormGroup } from '@angular/forms';
+
+@Component({
+  selector: 'ktb-time-input',
+  templateUrl: './ktb-time-input.component.html',
+  styleUrls: ['./ktb-time-input.component.scss'],
+})
+export class KtbTimeInputComponent implements OnInit {
+  @Input() required: boolean | undefined;
+  @Input() label = '';
+  @Input() secondsEnabled = true;
+  @Input() millisEnabled = true;
+  @Input() microsEnabled = true;
+  @Input() min?: Timeframe;
+  @Input() max?: Timeframe;
+  @Input() timeframe: Timeframe | undefined;
+  @Output()
+  timeChanged: EventEmitter<Timeframe> = new EventEmitter<Timeframe>();
+
+  public isFocused = false;
+
+  public console = console;
+
+  public timeForm = new FormGroup({
+    hours: new FormControl(),
+    minutes: new FormControl(),
+    seconds: new FormControl(),
+    millis: new FormControl(),
+    micros: new FormControl(),
+  });
+
+  public ngOnInit(): void {
+    if (this.timeframe) {
+      this.timeForm.controls.hours.setValue(this.timeframe.hours ?? null);
```

Comment:
The issue is ... if you already had something typed in and then remove it the value returned by the form field will still be `null`.
In Angular 13 there is an options property introduced, that could handle this: https://angular.io/api/forms/FormControlOptions#initialValueIsDefault
Then it could be set with `new FormControl(undefined, {initalValueIsDefault: true})`. Until we update to Angular 13, I guess that this is the only way to handle this.
The other possibility would be to change the `Timeframe` type to have number | null as property types.